Guidance document fix #9 (#72)
apple/swift-migration-guide: Guide.docc/Guidance.md

> # Guidance

Reviewer: I think the title should be more specific. Most of the guide is guidance on specific topics. Maybe something like "Migration Strategy"?

Author: ha yeah

Get recommendations on how to proceed with migrating your project to the
Swift 6 language mode.

Enabling complete concurrency checking in a module can yield many data-race
safety issues reported by the compiler.
Hundreds, possibly even thousands of warnings are not uncommon.
When faced with a such a large number of problems,
especially if you are just beginning to learn about Swift's data isolation
model, this can feel insurmountable.

**Don't panic.**

Frequently, you'll find yourself making substantial progress with just a few
changes.
And as you do, your mental model of how the Swift concurrency system works
will develop just as rapidly.

## Strategy

> Of course, there is no one single approach that will work for all projects.

Reviewer 2: I think we tend to avoid "of course" and similar like "it's obvious that..." sine for someone it may not be of course but be a surprise etc, so to not make them feel back avoid that wording?

```suggestion
There is no one single approach that will work for all projects.
```

Author: Yes, I actually really dislike this kind of wording too. Not sure how I let that happen.

> But there is a general strategy that is often very effective.

Reviewer 2: Isn't this a single sentence with the previous bit? `There is no..., but`
```suggestion
, but there is a general strategy that is often very effective.
```

A recommended approach has three key steps:

- Select a module
- Enable complete checking with Swift 5
- Address warnings

This process will be inherently _iterative_.
Even a single change in one module can have a large impact on the state of the
project as a whole.

> ## Module Selection

Reviewer 2: ```suggestion
## Migrating from the outside-in
```

```suggestion
## Start with leaf modules
```

Or something like that?

It is often easiest to start with the outer-most root module in a project.
This, by definition, is not a depenency of any other module.
Changes here can only have local effects, making it easier to
keep your changes contained.

## Enable Complete Checking

> Don't move directly from Swift 5 with no checking to the Swift 6 language mode.

Reviewer 2: Maybe rephrase into 
```suggestion
You don't have to directly from Swift 5 with no checking to the Swift 6 language mode.
```
